src/cmake-tools.ts

```diff
@@ -1208,6 +1209,16 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
       return;
     }
     await this._setDefaultBuildTarget(target);
+    const drv = await this._cmakeDriver;
+    await this.reRegisterTaskProviderforNewTarget(drv, target);
```

Comment:
I wonder if we need a call to reRegister here (when the default target is changed) since we have a call in run build. Is there a scenario where a build does not execute that other call to reRegister and the call here (setDefaultTarget) would matter?

Reply:
Yes, we need to re-register here. The taskprovider is hard-coded for only one target in the code. Ideally, we have to have only one task provider and register it only once, and keep multiple tasks in it. I will open an issue to refactor the task provider.
https://github.com/microsoft/vscode-cmake-tools/issues/1656